### PR TITLE
Consider devices with heartbeats to be out of the pre-provisioning state

### PR DESCRIPTION
**Pull request by thgreasi**
Change-type: minor
See: https://www.flowdock.com/app/rulemotion/r-resinos/threads/3cl3JBefZ2rV51CrrGpOz5T478d
See: https://jel.ly.fish/pattern-heartbeating-devices-without-vpn-connectivity-show-configuring-when-turned-off-f19c956
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

### CODE REVIEW

File: src/features/devices/models/device-additions.ts [hunk 1]
```diff
@@ -59,8 +59,13 @@ export const addToModel = (
...
 	const isPreProvisioning: AndNode = [
 		'And',
-		isOverallOffline,
+		['Equals', ['ReferencedField', 'device', 'is online'], ['Boolean', false]],
 		['NotExists', ['ReferencedField', 'device', 'last connectivity event']],
+		[
+			'Equals',
+			['ReferencedField', 'device', 'api heartbeat state'],
+			['EmbeddedText', 'unknown'],
+		],
```
Comment (thgreasi) [inline src/features/devices/models/device-additions.ts:68]: This change really just does whatever the comment above says...
The reasoning being that, in order for any device to have any heartbeat status != unknown, it means that the supervisor has managed to reach the API at least once, which means that the device is not really any more in pre-provisioning state.

Initial PR that introduces the previous logic
See: https://github.com/balena-io/balena-api/pull/2657/files

File: src/features/devices/models/device-additions.ts [hunk 2]
```diff
@@ -59,8 +59,13 @@ export const addToModel = (
 	// so it's still provisioning.
 	const isPreProvisioning: AndNode = [
 		'And',
-		isOverallOffline,
+		['Equals', ['ReferencedField', 'device', 'is online'], ['Boolean', false]],
```
Comment (Page-) [inline src/features/devices/models/device-additions.ts:62]: Is this redundant since having any connectivity event info would be the real tell?

Comment (thgreasi) [reply src/features/devices/models/device-additions.ts:62]: It should probably be fine without it. I only really added it for dependent devices.

**Comment by thgreasi** [comment]
@balena-ci I self-certify!

**Pull request merged**